rp2040-hal/src/gpio.rs

```diff
@@ -95,7 +95,13 @@ macro_rules! gpio {
                 type Parts = Parts;
 
                 fn split(self, pads: pac::$PADSX, sio: pac::SIO, resets: &mut pac::RESETS) -> Parts {
-                    resets.reset.modify(|_, w| w.$gpiox().clear_bit());
+                    resets.reset.modify(|_, w| w.$gpiox().clear_bit().$padsx().clear_bit());
+                    while resets.reset_done.read().$gpiox().bit_is_clear() {
+                        cortex_m::asm::delay(10);
+                    }
+                    while resets.reset_done.read().$padsx().bit_is_clear() {
+                        cortex_m::asm::delay(10);
+                    }
```

Comment:
At some point we could move these also to a HAL wrapper. Maybe add a ToDo note:
```suggestion
                    // TODO: Implement Resets in the HAL
                    while resets.reset_done.read().$gpiox().bit_is_clear() {
                        cortex_m::asm::delay(10);
                    }
                    while resets.reset_done.read().$padsx().bit_is_clear() {
                        cortex_m::asm::delay(10);
                    }
```